### PR TITLE
Client goal-script

### PR DESCRIPTION
**Pull request by memeToasty**
pls review 🥰👉👈

### CODE REVIEW

File: client/goals.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+PINS = [GOAL_ONE_PIN, GOAL_TWO_PIN]
+
+DEBOUNCING = 1000
+
+def button_callback(channel):
+    if channel == GOAL_ONE_PIN:
+        goal_id = GOAL_ONE_ID
+    else:
+        goal_id = GOAL_TWO_ID
+    url = f"http://{BACKEND_URL}/e/tor/{KICKER_ID}/{goal_id}"
```
Comment (TomRomeo) [inline client/goals.py:29]: add protocol to `BACKEND_URL`

File: client/goals.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        GPIO.add_event_detect(pin,GPIO.RISING,callback=button_callback, bouncetime=DEBOUNCING)
+
+
+args = parser.parse_args()
+BACKEND_URL = args.host
+KICKER_ID = args.kid
+setup()
+# perma loop; optimizable I guess
+while True:
+    continue
```
Comment (TomRomeo) [inline client/goals.py:51]: move this to 

```python
if __name__ == "__main__":
    # ...
```

**Review by TomRomeo — CHANGES_REQUESTED** [review]

File: client/goals.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import RPi.GPIO as GPIO
+import requests
+import argparse
+import asyncio
+
+# Argparsing
+parser = argparse.ArgumentParser(
+    description='Setting up goal counter for kicker table')
+parser.add_argument('-backend-url', type=str,
+                    default='https://localhost', help='URL of Pinguin-backend')
```
Comment (darmiel) [inline client/goals.py:10]: make the default `http://localhost `

File: README.md [hunk 4]
```diff
@@ -1,3 +1,19 @@
...
+
+#### Execution
+```
+usage: goals.py [-h] [-backend-url BACKEND_URL] [-kid KID]
+
+Setting up goal counter for kicker table
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -backend-url BACKEND_URL
```
Comment (darmiel) [inline README.md:16]:
```suggestion
  --backend-url BACKEND_URL
```

File: client/goals.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import RPi.GPIO as GPIO
+import requests
+import argparse
+import asyncio
+
+# Argparsing
+parser = argparse.ArgumentParser(
+    description='Setting up goal counter for kicker table')
+parser.add_argument('-backend-url', type=str,
```
Comment (darmiel) [inline client/goals.py:9]:
```suggestion
parser.add_argument('--backend-url', type=str,
```

**Review by darmiel — CHANGES_REQUESTED** [review]

File: client/goals.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+
+BACKEND_URL = "https://localhost"
+KICKER_ID = 0
+
+GOAL_ONE_PIN = 10
+GOAL_TWO_PIN = 12
+
+GOAL_ONE_ID = 0
+GOAL_TWO_ID = 1
```
Comment (darmiel) [inline client/goals.py:21]:
```suggestion
GOAL_BLACK_ID = 1
GOAL_WHITE_ID = 2
```

File: client/goals.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+parser.add_argument('--backend-url', type=str,
+                    default='http://localhost', help='URL of Pinguin-backend')
+parser.add_argument('--kid', type=str, default='0', help='ID of kicker table')
+
+
+BACKEND_URL = "https://localhost"
+KICKER_ID = 0
+
+GOAL_ONE_PIN = 10
+GOAL_TWO_PIN = 12
```
Comment (darmiel) [inline client/goals.py:18]:
```suggestion
GOAL_BLACK_ID = 10 # ?
GOAL_WHITE_ID = 12 # ?
```

File: client/goals.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+PINS = [GOAL_ONE_PIN, GOAL_TWO_PIN]
+
+DEBOUNCING = 500
+
+
+def button_callback(channel):
+    if channel == GOAL_ONE_PIN:
+        goal_id = GOAL_ONE_ID
+    else:
+        goal_id = GOAL_TWO_ID
```
Comment (darmiel) [inline client/goals.py:32]:
```suggestion
    if channel == GOAL_BLACK_PIN:
        goal_id = GOAL_WHITE_ID
    else:
        goal_id = GOAL_BLACK_ID
```

**Review by darmiel — CHANGES_REQUESTED** [review]

File: README.md [hunk 9]
```diff
@@ -1,3 +1,19 @@
...
 
-AA
+
+## Client
+
+### goal-tracking client
+
+#### Execution
+```
+usage: goals.py [-h] [--backend-url BACKEND_URL] [-kid KID]
```
Comment (TomRomeo) [inline README.md:10]:
```suggestion
usage: goals.py [-h] [--backend-url BACKEND_URL] [--kid KID]
```

File: client/goals.py [hunk 10]
```diff
@@ -0,0 +1,57 @@
+import RPi.GPIO as GPIO
+import requests
+import argparse
+import asyncio
+
+# Argparsing
+parser = argparse.ArgumentParser(
+    description='Setting up goal counter for kicker table')
+parser.add_argument('--backend-url', type=str,
+                    default='http://localhost', help='URL of Pinguin-backend')
```
Comment (darmiel) [inline client/goals.py:10]:
```suggestion
                    default='http://localhost', help='URL of Penguin')
```

**Review by darmiel — CHANGES_REQUESTED** [review]

**Review by darmiel — APPROVED** [review]

**Pull request merged**